ONSdigital/dis-authentication-stub
# Refactor Token Provisioning

### What

- Fixed linting issues
- Added CI
- Refactored to generate keys on start up, removing the need for key files
- Refactored static content behind an interface so it can be mocked more easily

### How to review

Check tests pass (now in CI)
To test JWT validation and issuing you can:

Start your services in order:
- dis-authentication-stub
- elasticsearch 2.4.2
- zebedee (pointing `IDENTITY_API_URL` at the stub and ensuring `ENABLE_JWT_SESSIONS` is true)
- dp-api-router (pointing `IDENTITY_API_URL` at the stub)

First test is that `zebedee` starts correctly (it won't if the JWK is invalid)

Second test is that the JWTs are issued and signed correctly:
- go to localhost:29500/florence/login and pick a user
- now visit localhost:29500/api/data

This now sends your token to zebedee which will parse and validate it - if it is successful, it should ask for a `uri`. Otherwise it will fail. 

### Who can review

Not me.
